### PR TITLE
[Fix](inverted index) fix memory leak in inverted index when encountering fault

### PR DESCRIPTION
**Pull request by airborne12**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by airborne12** [comment]
run buildall

**Comment by airborne12** [comment]
run buildall

File: be/src/olap/rowset/segment_v2/inverted_index_writer.cpp [hunk 1]
```diff
@@ -126,6 +126,12 @@ class InvertedIndexColumnWriterImpl : public InvertedIndexColumnWriter {
         }
     }
 
+    void close_on_error() override {
+        if (_index_writer) {
+            _index_writer->close();
```
Comment (qidaye) [inline be/src/olap/rowset/segment_v2/inverted_index_writer.cpp:131]: `close()` will write idx file and temporary directory. These files should be deleted too.

**Comment by airborne12** [comment]
run buildall

**Comment by airborne12** [comment]
run buildall

**Comment by airborne12** [comment]
run buildall

**Review by qidaye — APPROVED** [review]
LGTM

**Review by zzzxl1993 — APPROVED** [review]
LGTM

**Review by eldenmoon — APPROVED** [review]
LGTM

**Pull request merged**